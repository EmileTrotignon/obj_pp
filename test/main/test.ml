let () = Obj_pp.print_magic [4; 5; 6; 7]

let () = Obj_pp.print_magic (4, (5, (6, 7)))

let () = Obj_pp.print_magic [|4; 5; 6; 7|]

type adt = Int of int | Float of float | Cons of string * adt * adt

let my_adt =
  Cons ("second string", Cons ("first string", Int 12, Float 1.2), Float 1.4)

let my_adt = Cons ("more !", my_adt, my_adt)

let my_adt = Cons ("more !", my_adt, my_adt)

let my_adt = Cons ("more !", my_adt, my_adt)

let _f () x = x + 14

let () =
  let _x = 27 in
  Obj_pp.print_magic (_x, _f)

let () =
  let _x = 27 in
  let g y = _f () _x + y in
  Obj_pp.print_magic g

let () = Obj_pp.print_magic my_adt

let () = Obj_pp.print_magic None

let () = Obj_pp.print_magic [|1.2; 1.4; 1.6; 1.9|]
