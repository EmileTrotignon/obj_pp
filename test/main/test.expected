0 { 4; 0 { 5; 0 { 6; 0 { 7; 0 } } } }
0 { 4; 0 { 5; 0 { 6; 7 } } }
0 { 4; 5; 6; 7 }
0 { 27; <closure2 0x55d359edc4e0> }
<closure1 0x55d359edcae0>
2 { "more !"
  ; 2 { "more !"
      ; 2 { "more !"
          ; 2 { "second string"; 2 { "first string"; 0 { 12 }; 1 { 1.2 } }; 1 { 1.4 } }
          ; 2 { "second string"; 2 { "first string"; 0 { 12 }; 1 { 1.2 } }; 1 { 1.4 } } }
      ; 2 { "more !"
          ; 2 { "second string"; 2 { "first string"; 0 { 12 }; 1 { 1.2 } }; 1 { 1.4 } }
          ; 2 { "second string"; 2 { "first string"; 0 { 12 }; 1 { 1.2 } }; 1 { 1.4 } } } }
  ; 2 { "more !"
      ; 2 { "more !"
          ; 2 { "second string"; 2 { "first string"; 0 { 12 }; 1 { 1.2 } }; 1 { 1.4 } }
          ; 2 { "second string"; 2 { "first string"; 0 { 12 }; 1 { 1.2 } }; 1 { 1.4 } } }
      ; 2 { "more !"
          ; 2 { "second string"; 2 { "first string"; 0 { 12 }; 1 { 1.2 } }; 1 { 1.4 } }
          ; 2 { "second string"; 2 { "first string"; 0 { 12 }; 1 { 1.2 } }; 1 { 1.4 } } } } }
0
{| 1.2; 1.4; 1.6; 1.9 |}
